contracts/misc/LensPeriphery.sol

```diff
@@ -36,59 +36,60 @@ contract LensPeriphery {
 
     mapping(address => uint256) public sigNonces;
 
-    mapping(address => mapping(uint256 => string)) internal _metadataByProfileByOwner;
+    mapping(uint256 => string) internal _metadataByProfile;
 
     constructor(ILensHub hub) {
         HUB = hub;
     }
 
     /**
-     * @notice Sets profile metadata for a profile owner as a dispatcher.
+     * @notice Sets profile metadata for a profile as its dispatcher.
      *
      * @param profileId The profile ID to set the metadata for.
-     * @param metadata The metadata string to set for the profile and owner.
+     * @param metadata The metadata string to set for the profile.
      */
     function dispatcherSetProfileMetadataURI(uint256 profileId, string calldata metadata) external {
-        address owner = IERC721Time(address(HUB)).ownerOf(profileId);
         if (msg.sender != HUB.getDispatcher(profileId)) revert Errors.NotDispatcher();
-        _setProfileMetadataURI(owner, profileId, metadata);
+        _setProfileMetadataURI(profileId, metadata);
     }
 
     /**
-     * @notice Sets the profile metadata for a given profile when owned by the message sender.
+     * @notice Sets the profile metadata for a given profile.
      *
      * @param profileId The profile ID to set the metadata for.
-     * @param metadata The metadata string to set for the profile and message sender.
+     * @param metadata The metadata string to set for the profile.
      */
     function setProfileMetadataURI(uint256 profileId, string calldata metadata) external {
-        _setProfileMetadataURI(msg.sender, profileId, metadata);
+        address owner = IERC721Time(address(HUB)).ownerOf(profileId);
+        if (msg.sender != owner) revert Errors.NotProfileOwner();
+        _setProfileMetadataURI(profileId, metadata);
     }
```

Comment:
Maybe for consistency with LensHub makes sense to unify them into a single `setProfileMetadataURI` function that does the validation ensuring the sender is the owner or its dispatcher. Only disadvantage I see is a bit more of gas.

Reply:
Done in https://github.com/aave/lens-protocol/pull/112/commits/0d72b2fffcee19783de6e86e5c5a407bf388d208!